File: removable/event/p_event_form.erl
-module(p_event_form).

-compile(export_all).

-include("sticky.hrl").
-include("event.hrl").

render(Method, Action, Event, Tags) ->
    Js = ale:ff("p_event_form.js"),
    ale:app_add_js(Js),

    Cancel = case Method of
        put -> [{a, [{href, ale:path(event, show, [Event#event.id])}], ?T("Cancel")}, " "];
        _   -> ""
    end,
    {form, [{id, event_form}, {method, post}, {action, Action}], [
        {input, [{type, hidden}, {name, "_method"}, {value, Method}]},

        {span, [{class, label}], ?T("Name")},
        {input, [{type, text}, {class, textbox}, {name, name}, {value, Event#event.name}]},

        {span, [{class, label}], ?T("Invitation")},
        {textarea, [{name, invitation}], Event#event.invitation},

        {span, [{class, label}], ?T("Registration deadline")},
        {input, [{type, text}, {class, "textbox quarter date_picker"}, {name, deadline_on}, {value, h_application:render_date(Event#event.deadline_on)}]},

        h_tag:render_tag_selection(Tags),

        h_application:render_mathcha(),

        {input, [{type, submit}, {class, button}, {value, ?T("Save")}]}, " ", Cancel
    ]}.

File: removable/event/p_event_form.js
$('#event_form input[type=submit]').click(function() {
    $(this).hide();
    $(this).after('<img class="ajax-loader" src="/static/img/spinner.gif" />');

    tinyMCE.triggerSave();

    var action     = $('#event_form').attr('action');
    var _method    = $('#event_form input[name=_method]').val();
    var name       = $('#event_form input[name=name]').val();
    var invitation = $('#event_form textarea[name=invitation]').val();
    var deadlineOn = $('#event_form input[name=deadline_on]').val();
    var tags       = $('#event_form input[name=tags]').val();
    var answer     = $('#event_form input[name=answer]').val();
    var eAnswer    = $('#event_form input[name=encrypted_answer]').val();
    var postData   = {_method: _method, name: name, invitation: invitation, deadline_on: deadlineOn, tags: tags, answer: answer, encrypted_answer: eAnswer};

    $.post(action, postData, function(data) {
        if (data.error) {
            alert(data.error);
            $('#event_form .ajax-loader').remove();
            $('#event_form input[type=submit]').show();
        } else {
            window.location.href = "/events/" + data.atomic;
        }
    }, "json");
    return false;
});
